src/modules/heroku.rs

```diff
@@ -0,0 +1,501 @@
+            if apps.len() == 1 {
+                return Some(apps[0].clone());
+            }
+            // there can be more than one valid app, in more than one git remote.
+            // In Heroku commands you typically have to provide `--remote` to
+            // define which remote (and through that the app) to use.
+            // Here we don't show any app when there is more than one.
+        }
+
+        None
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use super::*;
+        use crate::{
+            test::{default_context, fixture_repo, FixtureProvider},
+            utils::create_command,
+        };
+        use std::io;
+        use tempfile::TempDir;
+
+        fn add_remote(repo_dir: &TempDir, name: &str, url: &str) -> io::Result<()> {
+            create_command("git")?
+                .args(&["remote", "add", name, url])
+                .current_dir(repo_dir.path())
+                .output()?;
+
+            Ok(())
+        }
+
+        #[test]
+        fn everything_empty() {
+            let context = default_context();
+            assert!(get_app(&context).is_none());
+        }
+
+        #[test]
+        fn from_env() {
+            let mut context = default_context();
+            context.env.insert("HEROKU_APP", "some_app".to_owned());
+            assert_eq!(get_app(&context).unwrap(), "some_app");
+        }
+
+        #[test]
+        fn from_remote() -> io::Result<()> {
+            let repo_dir = fixture_repo(FixtureProvider::Git)?;
+            let mut context = default_context();
+            context.current_dir = repo_dir.path().into();
+
+            add_remote(&repo_dir, "some_name", "git@heroku.com:some_app_name.git")?;
+            add_remote(&repo_dir, "some_other_name", "git@github.com:repo.git")?;
+
+            assert_eq!(get_app(&context).unwrap(), "some_app_name");
+
+            Ok(())
+        }
+
+        #[test]
+        fn multiple_apps() -> io::Result<()> {
+            let repo_dir = fixture_repo(FixtureProvider::Git)?;
+            let mut context = default_context();
+            context.current_dir = repo_dir.path().into();
+
+            add_remote(&repo_dir, "some_name", "git@heroku.com:some_app_name.git")?;
+            add_remote(
+                &repo_dir,
+                "some_other_name",
+                "https://git.heroku.com/other_app_name.git",
+            )?;
+
+            assert!(get_app(&context).is_none());
+
+            Ok(())
+        }
+    }
+}
+
+fn account_from_netrc_file(context: &Context) -> Option<String> {
+    // https://linux.die.net/man/5/netrc
+    // This parses simple .netrc files, it could return wrong
+    // results for files with `macdef`.
+    // We only look at explicit machine definitions, not at
+    // defaults. Heroku doesn't use the netrc default section.
+    // Lastly, Heroku uses the `netrc-parser` package to edit
+    // the file, which also supports GPG encrypted netrc files.
+    // This we also don't support right now.
+
```

Comment:
This won't catch all edge-cases, is this OK? 

Alternative would be to write a small PEST grammar, or use the [netrc lib](docs.rs/netrc/) (while it wasn't updated in a while)